server/src/test/java/keywhiz/service/resources/admin/SecretsResourceTest.java

```diff
@@ -99,13 +105,54 @@
               NOW, "creator", NOW, "updater",
               emptyMap, 0));
 
-  AuditLog auditLog = new SimpleLogger();
-
   SecretsResource resource;
 
   @Before
   public void setUp() {
     resource = new SecretsResource(secretController, aclDAO, secretDAO, auditLog);
+
+    when(secretController.builder(any(), any(), any(), anyLong()))
+        .thenReturn(secretBuilder);
+    when(secretBuilder.withDescription(any())).thenReturn(secretBuilder);
+    when(secretBuilder.withMetadata(any())).thenReturn(secretBuilder);
+    when(secretBuilder.withOwnerName(any())).thenReturn(secretBuilder);
+    when(secretBuilder.withType(any())).thenReturn(secretBuilder);
+    when(secretBuilder.create()).thenReturn(secret);
+    when(secretBuilder.createOrUpdate()).thenReturn(secret);
+
+    when(secretController.getSecretById(secret.getId())).thenReturn(Optional.of(secret));
+  }
+
+  @Test
+  public void createsSecretWithOwner() {
+    String owner = UUID.randomUUID().toString();
+
+    CreateSecretRequestV2 request = CreateSecretRequestV2.builder()
+        .content(secret.getSecret())
+        .owner(owner)
+        .name(secret.getName())
+        .build();
+
+    resource.createSecret(user, request);
+
+    verify(secretBuilder).withOwnerName(owner);
+  }
```

Comment:
This is the test that verifies that the bug is fixed (at least at the unit test level, there's a new integration test that also provides more comprehensive coverage).